mozilla-services/antenna
# bug 1723964: update everett, redo config, create crashmover

This refactors a lot of connective tissue in Antenna as a result of updating Everett to 2.0.0 which picks up the new component configuration code:

* update everett to 2.0.0
* move the crashmoving code out of the breakpad_resource and into a new crashmover module; we could use the crashmover for other payload types--it's not specific to the breakpad style crash report
* since component configuration namespaces reflect the component hierarchy, that caused some namespaces to change, so we had to update that
* the docs around configuration were a mess, so I improved those and rewrote a lot of bits
* the way logging and metrics were set up was clunky, so I adopted the style I used with Eliot so it's easier to follow over interface boundaries

To test:

1. run antenna and verify the logs contain the correct configuration names and values for the local dev environment
2. run unit tests
3. run antenna and in another shell run the systemtests
